A2C Rev2 Schematic Review

ryan-and opened the pull request:
**Schematic Review (A2C Rev2)**
Purpose:

In tandem with the debugger PCB this board will allow us to begin our transition to an STM32 based platform. Depending on whether sensors use UART, SPI, or I2C, future revisions can be made to reduce the form factor and increase the number of available sensor inputs.

Added:

- I2C
- SPI
- LDO

Commit:
Analog to CAN Sch

ryan-and committed:
Started rev2

ryan-and committed:
LDO added

ethanmpeterson reviewed (approved):
**Schematic Checklist**
- All important nets named **YES**
- No spaces in net names (use underscores) **YES**
- Net names do NOT begin with numbers or other special characters such as +-/\. This makes naming software variables corresponding to a certain net easier. **NO (12V power net is called +12V, fine in this case since we would never enable / disable the rail in software)**
- No 4 point connections **YES**
- Short circuit protection via current monitor implementation or fuse, Specifically for power rails supplying expensive parts **N/A**
- No circuit compilation errors **YES**
- Active low signals have _n at the end of their net name **N/A**
- Differential pair nets are named with _P and _N endings. Both nets in the pair should also be marked with the differential pair schematic marking in Altium **YES**

**General Feedback**
- The LM3940 LDO you have has a max input of 5.5 V making it unsuitable to regulate the 12V from the car. One solution would be to add a regulator in series that steps the 12 V down to 5 V. A regulator that JLC PCB has in stock is this [one](https://datasheet.lcsc.com/szlcsc/Texas-Instruments-TI-LM2940CSX-5-0-NOPB_C133781.pdf) from TI. Also remember to add the input and output capacitors specified in the datasheet as the current LDO in the schematic is missing them.
- I think we should also have a few more decoupling caps for the MCU. Datasheet calls a 4.7uF cap to ground on VDD_3. Also need a 10nF + 1uF (parallel) connected between VDDA and GND. Power supply scheme from datasheet is given below.
![image](https://user-images.githubusercontent.com/16881963/103377437-6937e300-4aad-11eb-8ae2-68fb1d978cb1.png)
- The UART pins selected are correct
- SPI pins are correct
- CAN pins are correct
- SWD pins are correct
- I2C Pins are correct

This looks awesome! I am going to approve this, just be sure to make the schematic tweaks above before proceeding to layout.

Ethan

ryan-and committed:
LDO and MCU caps

ryan-and commented:
I changed the LDO to the one I specced out before (higher max voltage input) and I believe I added the correct caps. I also added the caps to the VDDA and VDD_3. Would be good to double check that I did that right though

ethanmpeterson committed:
add VCC3V3 connection

ryan-and committed:
A2C rev2 layout

ryan-and committed:
PCB meets design rules check

ryan-and committed:
Rev2 complete
